rafu.js

```diff
@@ -8,38 +8,66 @@
 /**
  * rafu.frame - basic animation frame request method
  * @param {Function} callback
+ * @return {Number} requestAnimationFrame or setTimeout id for cancelling
  */
 var frame = exports.frame = (function(win) {
   var raf = win.requestAnimationFrame ||
     win.webkitRequestAnimationFrame ||
     win.mozRequestAnimationFrame ||
     win.oRequestAnimationFrame ||
     win.msRequestAnimationFrame;
-  if (raf) return raf.bind(win);
+  if (raf) {
+    return raf.bind(win);
+  }
   return function (callback) {
-    setTimeout(callback, 16);
+    return setTimeout(callback, 16);
+  };
+}(typeof window === 'undefined' ? {} : window));
+
+/**
+ * rafu.cancelFrame - cancels a frame via id returned by rafu.frame
+ * @param {Number} requestAnimationFrame or setTimeout id for cancelling
+ */
+var cancelFrame = exports.cancelFrame = (function(win) {
+  var cancelRaf = win.cancelAnimationFrame ||
+    win.webkitCancelAnimationFrame ||
+    win.mozCancelAnimationFrame ||
+    win.oCancelAnimationFrame ||
+    win.msCancelAnimationFrame;
+  if (cancelRaf) {
+    return cancelRaf.bind(win);
+  }
+  return function (timeoutId) {
+    return win.clearTimeout(timeoutId);
   };
 }(typeof window === 'undefined' ? {} : window));
 
 /**
  * rafu.throttle - returns a function, that, when invoked, will only be triggered
  * once every browser animation frame
  * @param {Function} func  Funciton to wrap
- * @return {Funciton}
+ * @return {Function} func with cancelFrame property added  
  */
 exports.throttle = function(func) {
   var wait;
   var args;
   var context;
+  var rafOrTimeoutId;
 
-  return function() {
+  var throttled = function() {
     args = arguments;
     if (wait) return;
     wait = true;
     context = this;
-    frame(function() {
+    rafOrTimeoutId = frame(function() {
       wait = false;
       func.apply(context, args);
     });
   };
+
+  throttled.cancelFrame = function() {
```

Comment:
nit: rename this to just `cancel`. Lodash seems to provide both `cancel` and `flush` methods on the returned function.

https://lodash.com/docs/4.17.11#debounce

![Screen Shot 2019-06-03 at 4 35 07 PM](https://user-images.githubusercontent.com/227879/58832739-998f7500-861d-11e9-810c-f6148d2a2adc.png)

Reply:
I mirrored `rafu.cancelFrame`... should they both be `cancel`?

Reply:
Nevermind, `throttledFunc.cancel()` would make sense.

Reply:
I’d personally just call them both `cancel`.

Reply:
Changed name and added test that it's there: https://github.com/danro/rafu/pull/2/commits/254e3814bcc9ecd2310c4c380afd617023233cb9

Reply:
Meh... `cancel(thing)` seems a little generic. 🤷‍♀

Reply:
I like `throttled.cancel()` as well. Generic isn't bad in this case, we simply want the throttled function to cancel the execution (without caring too much about the implementation details).

Reply:
`rafu.cancel` also seems more in-line with the simplicity of the module. If we supported other things besides frames then maybe the distinction would matter.